org.mixedrealitytoolkit.input/Interactors/GazePinch/GazePinchInteractor.cs

```diff
@@ -444,5 +452,64 @@ protected override void OnSelectExited(SelectExitEventArgs args)
         }
 
         #endregion XRBaseInteractor
+
+        /// <summary>
+        /// Updates the pinch state of the GazePinchInteractor.
+        /// If a HandController is present, the pinch state is updated using the HandController's HandNode defaulting to XRNode.RightHand if not set.
+        /// If no HandController is present, the pinch state is updated using the Handedness of the GazePinchInteractor defaulting to InteractorHandedness.Right if not set.
+        /// If the pinch data is not available for the set hand then the other hand is tried.
+        /// </summary>
+        private void UpdatePinchState()
+        {
+            if (logicalSelectState == null)
+            {
+                Debug.LogWarning("GazePinchInteractor is missing logicalSelectState, pinch state won't update.");
+                return;
+            }
+
+            if (XRSubsystemHelpers.HandsAggregator == null)
+            {
+                return;
+            }
+
+            // Determine the XRNode to use for the pinch data
+            bool gotPinchData;
+            XRNode xRNode;
+            if (handController != null) //For pre-XRI3 code.  Note: This if can be removed (leave the code in the 'else' section) when
+                                        //Controllers are fully migrated to XRI 3 era in which all Controllers are removed.
+            {
+                xRNode = handController.HandNode;
+                if (xRNode != XRNode.LeftHand && xRNode != XRNode.RightHand)
+                {
+                    Debug.LogWarning($"HandController {handController.name} does not have HandNode set to neither XRNode.LeftHand nor XRNode.RightHand, defaulting to XRNode.RightHand.");
+                    xRNode = XRNode.RightHand;
+                }
+            }
+            else //For post-XRI3 code.  The XRNode is determined by the GazeInteractor Handedness if none is set then it defaults to XRNode.RightHand.
+            {
+                switch (handedness)
```

Comment:
make a helper extension for this.  And put in HandednessExtensions.cs

Reply:
I wasn't able to add to HandednessExtensions because GazePinchInteractor's handedness is of type InteractorHandedness and not of type Handedness.  So I created a new InteractorHandednessExtensions class and added it there.  Changes in https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/743/commits/6b3d24f5e3c85dd9d95dbc08721409bf828c1b96 , resolving.